### PR TITLE
[Security] Initialize  Security Autogenerated Modules

### PR DESCRIPTION
**Pull request by ivadim**
## Description

This PR adds the commands for api-collections and security-connectors APIs
Covering next REST specs:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/security/resource-manager/Microsoft.Security/preview/2023-09-01-preview/securityConnectorsDevOps.json
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/security/resource-manager/Microsoft.Security/preview/2023-10-01-preview/securityConnectors.json
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/security/resource-manager/Microsoft.Security/stable/2023-11-15/apiCollections.json

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by ivadim** [comment]
Adding docs and examples while waiting design review https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1411

File: src/Security/Security.Autorest/examples/Get-AzSecurityApiCollection.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
+```powershell
```
Comment (msJinLei) [inline src/Security/Security.Autorest/examples/Get-AzSecurityApiCollection.md:2]: Please provide the example for the cmdlet. The same below

File: src/Security/Security.Autorest/docs/Get-AzSecurityConnectorAzureDevOpsRepos.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
+---
+external help file:
+Module Name: Az.Security
+online version: https://learn.microsoft.com/powershell/module/az.security/get-azsecurityconnectorazuredevopsrepos
+schema: 2.0.0
+---
+
+# Get-AzSecurityConnectorAzureDevOpsRepos
```
Comment (msJinLei) [inline src/Security/Security.Autorest/docs/Get-AzSecurityConnectorAzureDevOpsRepos.md:8]: Please use singular name for the cmdlet https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#specific-noun-and-noun-singularity

File: src/Security/Security.Autorest/docs/Get-AzSecurityConnectorGitHubRepos.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
+---
+external help file:
+Module Name: Az.Security
+online version: https://learn.microsoft.com/powershell/module/az.security/get-azsecurityconnectorgithubrepos
+schema: 2.0.0
+---
+
+# Get-AzSecurityConnectorGitHubRepos
```
Comment (msJinLei) [inline src/Security/Security.Autorest/docs/Get-AzSecurityConnectorGitHubRepos.md:8]: Please use the singular name for the cmdlet Get-AzSecurityConnectorGitHubRepos

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Security/Security.Autorest/docs/Get-AzSecurityConnectorAzureDevOpsRepos.md (hunk 2 above)
Comment (ivadim) [reply src/Security/Security.Autorest/docs/Get-AzSecurityConnectorAzureDevOpsRepos.md]: Thank you for the review. I made all names singular.

File: src/Security/Security.Autorest/examples/Get-AzSecurityApiCollection.md (hunk 1 above)
Comment (ivadim) [reply src/Security/Security.Autorest/examples/Get-AzSecurityApiCollection.md:2]: Thank you for the review. I added examples for all cmdlets.

**Comment by msJinLei** [comment]
Not use a normal operationId
https://github.com/Azure/azure-rest-api-specs/blob/ff9fb162a8a79531058ec5c43c49169c92311717/specification/security/resource-manager/Microsoft.Security/preview/2023-09-01-preview/securityConnectorsDevOps.json#L19

**Pull request merged**